src/ws/cockpitauth.c

```diff
@@ -307,6 +315,32 @@ type_option (const gchar *type,
   return default_str;
 }
 
+static guint
+timeout_option (const gchar *type)
+{
+  guint timeout = cockpit_ws_auth_process_timeout;
+  guint64 conf_timeout;
+  const gchar* conf = type_option (type, "timeout", NULL);
+
+  if (conf)
+    {
+      conf_timeout = g_ascii_strtoull (conf, NULL, 10);
+      if (errno == ERANGE || errno == EINVAL)
+        timeout = cockpit_ws_auth_process_timeout;
+      else if (conf_timeout > MAX_AUTH_TIMEOUT || conf_timeout > UINT_MAX)
+        timeout = (MAX_AUTH_TIMEOUT > UINT_MAX) ? UINT_MAX : MAX_AUTH_TIMEOUT;
+      else if (conf_timeout < MIN_AUTH_TIMEOUT)
+        timeout = MIN_AUTH_TIMEOUT;
+      else
+        timeout = (uint)conf_timeout;
```

Comment:
Technically I think it would be safer to use guint here, since that how timeout is declared:
`timeout = (guint)conf_timeout;`